### PR TITLE
[SPARK-35684][INFRA][PYTHON] Bump up mypy version in GitHub Actions

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to bump up the mypy version to 0.910 which is the latest.

### Why are the changes needed?

To catch the type hint mistakes better in PySpark.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

GitHub Actions should test it out.

### CODE REVIEW

**Review by srowen — APPROVED** [review]

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -366,9 +366,7 @@ jobs:
         #   See also https://github.com/sphinx-doc/sphinx/issues/7551.
         # Jinja2 3.0.0+ causes error when building with Sphinx.
         #   See also https://issues.apache.org/jira/browse/SPARK-35375.
-        # TODO(SPARK-35684, SPARK-35683): Bump up the mypy version. This is blocked by
-        #   Index.difference logic issue.
-        python3.6 -m pip install flake8 pydata_sphinx_theme 'mypy==0.812' numpydoc 'jinja2<3.0.0' 'black==21.5b2'
+        python3.6 -m pip install flake8 pydata_sphinx_theme 'mypy==0.910' numpydoc 'jinja2<3.0.0' 'black==21.5b2' types-tabulate
```
Comment (ueshin) [inline .github/workflows/build_and_test.yml:369]: I'd rather add the `tabulate` entry in `mypy.ini`.

```
[mypy-tabulate.*]
ignore_missing_imports = True
```

Comment (ueshin) [reply .github/workflows/build_and_test.yml:369]: Or we should install other libraries and remove the entries in `mypy.ini` (in the future)?

Comment (HyukjinKwon) [reply .github/workflows/build_and_test.yml:369]: Oh okay. I think missing import would be better approach

**Comment by HyukjinKwon** [comment]
Python linter passed. Let me get this in!

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.2.

**Pull request closed without merging**